### DISCUSSION TITLE
string type narrowing

### DISCUSSION THREAD

**Discussion by meadowsys** [colinhacks/zod#381] [Q&A] [answered]
hi!

I want to use zod (v3) to validate a string to be one of a few strings like this:

![image](https://user-images.githubusercontent.com/49562048/113523793-ecf6e500-955e-11eb-84d9-8753e014159b.png)

However, the type of that property (category) is still `string`:

![image](https://user-images.githubusercontent.com/49562048/113523807-04ce6900-955f-11eb-929b-99f7aa1a776c.png)

I found a way to get the right type out of it using a transform afterwards:

![image](https://user-images.githubusercontent.com/49562048/113523844-5d9e0180-955f-11eb-8453-2d2c98ab6494.png)

![image](https://user-images.githubusercontent.com/49562048/113523850-64c50f80-955f-11eb-987f-9d54890866a2.png)

This feels verbose and extremely repetitive though. Is there a better way to do this?

**Comment by prescience-data** [comment, marked as answer]
Is there a reason for not using `enum` ( which is what this is for )?

```typescript
z.object({
  category: z.enum(["block", "sound", "lang", "misc"]).default("block")
})
```

**Reply by meadowsys** [reply]
I think that's exactly what I needed. Thanks!

**Reply by colinhacks** [reply]
Thanks @prescience-data!! Really appreciate you chiming in on these discussions!
